common/views/components/NewsletterSignup/NewsletterSignup.js

```diff
@@ -0,0 +1,193 @@
+  },
+  {
+    id: 'youth_and_community_workers',
+    label: `Updates for Youth & Community Workers`,
+    name: 'addressbook_40133',
+    description: `Updates for Youth & Community Workers, featuring events and activities for youth 14-19`
+  }
+];
+
+class NewsletterSignup extends Component<Props, State> {
+  state = {
+    checkedInputs: [],
+    isEmailError: true,
+    isCheckboxError: true,
+    noValidate: false,
+    isSubmitAttempted: false
+  };
+
+  componentDidMount() {
+    this.setState({
+      noValidate: true
+    });
+  }
+
+  updateCheckedInputs = (event) => {
+    const isChecked = event.target.checked;
+    const id = event.target.id;
+
+    const newInputs = isChecked
+      ? this.state.checkedInputs.concat(id)
+      : this.state.checkedInputs.filter(c => c !== id);
+
+    this.setState({
+      checkedInputs: newInputs,
+      isCheckboxError: newInputs.length === 0
+    });
+  }
+
+  handleEmailInput = (event) => {
+    this.setState({
+      isEmailError: !event.target.validity.valid
+    });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    this.setState({
+      isSubmitAttempted: true
+    });
+
+    if (!this.state.isCheckboxError && !this.state.isEmailError) {
+      event.target.submit();
+    }
+  }
+
+  render() {
+    const { isConfirmed, isSuccess, isError } = this.props;
+    return (
+      <div className='body-text'>
+        {isConfirmed &&
+          <Fragment>
+            <h1>Confirmation message</h1>
+            <p><a href='/whats-on'>Browse our current and upcoming exhibitions and events</a>.</p>
+          </Fragment>
+        }
+
+        {isSuccess &&
+          <Fragment>
+            <h1>Thank you</h1>
+            <p>Thank you for your interest in receiving updates from Wellcome Collection.</p>
+            <p>If this is first time you have subscribed to a newsletter, you will receive an email asking to confirm your subscription. Please check your email to confirm and start receiving updates.</p>
+            <p><a href='/whats-on'>Browse our current and upcoming exhibitions and events</a>.</p>
+          </Fragment>
+        }
+
+        {isError &&
+          <Fragment>
+            <h1>There was a problem</h1>
+            <p>Please try again.</p>
+          </Fragment>
+        }
+
+        {!isConfirmed && !isSuccess && !isError &&
+          <h1>Sign up</h1>
+        }
+
+        {!isConfirmed && !isSuccess &&
+          <form
+            noValidate={this.state.noValidate}
+            onSubmit={this.handleSubmit}
+            name='newsletter-signup'
+            id='newsletter-signup'
+            action='https://r1-t.trackedlink.net/signup.ashx'
+            method='post'>
+            <input type='hidden' name='userid' value='225683' />
+            <input type='hidden' name='ReturnURL' value='http://localhost:3000/info/newsletter' />
+            <input type='hidden' name='SIG22a9ece3ebe9b2e10e328f234fd10b3f5686b9f4d45f628f08852417032dc990' value='' />
```

Comment:
?

Reply:
dotmailer needs this (and provided us the name)

Reply:
Fancy leaving a note as it looks like testing code?